syft/pkg/cataloger/golang/scan_bin.go

```diff
@@ -33,7 +33,7 @@ func scanFile(reader unionReader, filename string) ([]*debug.BuildInfo, []string
 	for _, r := range readers {
 		bi, err := buildinfo.Read(r)
 		if err != nil {
-			log.Warnf("golang cataloger: scanning file %s: %v", filename, err)
+			log.Debugf("golang cataloger: scanning file %s: %v", filename, err)
```

Comment:
So my opinion is that this **_should_** be a `WARN` log entry. If Syft knows for sure that it missed an opportunity to scan and surface a piece of the software, that's noteworthy (the same idea that's culminated in https://github.com/anchore/syft/issues/518). For example, it might be that the scanned Go binaries has no known vulnerabilities... except in the section that Syft missed. It's important to have a way to signal to the user the difference between "Everything is fine ✅ " and "Everything _that we scanned_ is fine, but we weren't able to scan everything. ⚠️ "

Reply:
> If Syft knows for sure that it missed an opportunity to scan and surface a piece of the software, that's noteworthy (the same idea that's culminated in https://github.com/anchore/syft/issues/518). For example, it might be that the scanned Go binaries has no known vulnerabilities... except in the section that Syft missed. It's important to have a way to signal to the user the difference between "Everything is fine ✅ " and "Everything that we scanned is fine, but we weren't able to scan everything. ⚠️ "

100% agree.

However, in this case this is a little bit grey. The golang cataloger selects candidate files based on MIME type (get all executables). There is no distinction of the binary being analyzed being a go-compiled binary (we don't know if it should be). Defaulting to warning here seems wrong since it seems to imply that we were expecting that all inputs are go-compiled binaries (which is not true). 

That being said, I also agree that we don't want to not report up skipping a file that was intended to be analyzed by the golang cataloger but was not. We should show warnings for these cases.

Given the above context, we should probably default to having this being a `debug` (though it does really seem like it should be warning at a first glance).

Reply:
I like the idea of debug `not a go executable` and warn on any other error. My design north here is: how would things look like once we add more binary catalogers? In this future scenario log messages of the type `not a X executable` are more noise than information, but we are not there yet.